src/metabase/sync/fetch_metadata.clj

```diff
@@ -22,3 +23,8 @@
   (let [driver (driver/->driver database)]
     (when (driver/driver-supports? driver :foreign-keys)
       (driver/describe-table-fks driver database table))))
+
+(s/defn ^:always-validate db-timezone :- s/Str
```

Comment:
It might be possible to define an output schema a little stricter than just `s/Str` here. You could do `su/NonBlankString` or even better define a new schema that checks whether the timezone is legit. You could do something like check whether `(java.util.TimeZone/getTimeZone <str>)` works correctly (for an invalid timezone it just returns `GMT` so you could check whether the `TimeZone` you get back isn't GMT, unless of course the string passed in was `"GMT"`, in which case that would be ok)